pages/components/UserInput/userInput.js

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect, useState, useRef } from 'react';
+import styles from '../../../styles/UserInput.module.css';
+import data from '../../../data/dictionary.json';
+import {useRouter} from 'next/router';
+
+function UserInput(props) {
+    const router =useRouter();
+    const {level} = router.query
+    const [userText, SetUserText] = useState('');
+    const [currentWord, setCurrentWord] = useState('')
+    const [difficulty, setDifficulty] = useState({ level: level, factor: 1});
+
+    const getWords = () => {
+       let easyWords = [],mediumWords = [],hardWords = [];
+
+       let difficultyfactor;
+       if (difficulty.level === 'easy') {
+           difficultyfactor = 1;
+       } else if (difficulty.level === 'medium') {
+           difficultyfactor = 1.5;
+       } else {
+           difficultyfactor = 2;
+       }
+
+       setDifficulty({ level: difficulty.level, factor: difficultyfactor });
+       var words = props.arrayofWords;
+       if(Object.keys(props.arrayofWords).length === 0){
+        for (const word of data) {
+            if (word.length <= 4) {
+                easyWords.push(word)
+            }
+            else if (word.length <= 8) {
+                mediumWords.push(word);
+            }
+            else{
+                hardWords.push(word);
+            }
+        }
```

Comment:
This will create a performance bottleneck as you are running loop on large set of data. In general this kind of data is pre processed during build time for static websites like this, I mean when your `dist` folder is generated this data.json should be broken down before hand to make your UI load faster.